index.js

```diff
@@ -10,41 +10,72 @@ const path = require("path");
 const loaderUtils = require("loader-utils");
 const SourceNode = require("source-map").SourceNode;
 const SourceMapConsumer = require("source-map").SourceMapConsumer;
-const FOOTER = "/*** REQUIRES FROM autoload-plugins-loader ***/\n";
+const matchAll = require('match-all');
 
-module.exports = function(content, sourceMap) {
-    const resourcePath = this.resourcePath;
+const FOOTER = "/*** REQUIRES FROM autoload-modules-loader ***/\n";
+
+const REQUIRE_RE = /\brequire\s*\(\s*[\'"]([^\'"]*)[\'"]\s*\)/g;
 
+module.exports = function(content, sourceMap) {
     if (this.cacheable) {
         this.cacheable();
     }
 
-    const query = loaderUtils.getOptions(this) || {};
-    const requires = [];
-    const keys = Object.keys(query);
-
-    if (keys.length == 1 && typeof query[keys[0]] == "boolean") {
-        const mod = keys[0];
-        const modPath = path.resolve(__dirname, mod);
-        requires.push(`require("${modPath}");   // ${mod}`);
-    } else {
-        keys.forEach(function(name) {
-            const mod = name;
-            const modPath = path.resolve(__dirname, mod);
-            requires.push(`require("${modPath}");   // ${mod}`);
+    // promisify-node doesn't know how to deal with methods so we have to wrap
+    // this.resolve() ourselves.
+    const resolve = (context, request) =>
+        new Promise((resolve, reject) => {
+            this.resolve(context, request, (err, result) => {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(result);
+                }
+            });
         });
-    }
 
-    if (sourceMap) {
-        const currentRequest = loaderUtils.getCurrentRequest(this);
-        const node = SourceNode.fromStringWithSourceMap(content, new SourceMapConsumer(sourceMap));
-        node.add("\n\n" + FOOTER + requires.join("\n"));
-        const result = node.toStringWithSourceMap({
-            file: currentRequest
-        });
-        this.callback(null, result.code, result.map.toJSON());
-        return;
-    }
+    const reqs = matchAll(content, REQUIRE_RE).toArray();
+    const reqsToAdd = [];
+
+    // Use async mode b/c we calling this.resolve() is async
+    const callback = this.async();
+
+    Promise.all(reqs.map(req => resolve(this.context, req))).then(resolvedReqs => {
+        const requires = [];
+        const {moduleToPluginsMap} = this.query;
+
+        for (const resolvedReq of resolvedReqs) {
+            if (resolvedReq in moduleToPluginsMap) {
```

Comment:
Check if any of the resolved require statement paths match the keys in `modulesToPluginsMap`.